src/runtime/virtcontainers/hypervisor.go

```diff
@@ -742,23 +737,32 @@ func RunningOnVMM(cpuInfoPath string) (bool, error) {
 		}
 
 		for _, field := range fields[1:] {
-			if field == "hypervisor" {
-				return true, nil
-			}
+			flags[field] = true
 		}
 
-		// As long as we have been able to analyze the fields from
-		// "flags", there is no reason to check what comes next from
-		// /proc/cpuinfo, because we already know we are not running
-		// on a VMM.
-		return false, nil
+		return flags, nil
 	}
 
 	if err := scanner.Err(); err != nil {
+		return map[string]bool{}, err
+	}
+
+	return map[string]bool{}, fmt.Errorf("Couldn't find %q from %q output", flagsField, cpuInfoPath)
+}
+
+// RunningOnVMM checks if the system is running inside a VM.
+func RunningOnVMM(cpuInfoPath string) (bool, error) {
+	if runtime.GOARCH == "arm64" || runtime.GOARCH == "ppc64le" || runtime.GOARCH == "s390x" {
```

Comment:
While we are at it, could we turn this test into a positive test for the supported scenarios (i.e. list the architectures where we know, rather than the architectures where we don't know)? That would help identify the targets we believe should be tested.